### ISSUE TITLE
Incorrect MLIL translation of ConstPointer load?

### ISSUE THREAD

**Issue by yrp604** [Vector35/binaryninja-api#2303]
I have a small function that looks up a thread local storage structure based on an argument:

```
tls_for_core:
0e035684  010880d2   mov     x1, #0x40
0e035688  007c019b   mul     x0, x0, x1
0e03568c  a1830a10   adr     x1, 0xe04a700
0e035690  0000018b   add     x0, x0, x1
0e035694  c0035fd6   ret     
```

0xe04a700 is the base the the TLS array. The first curious thing is it's not displaying the address with the symbol name that's defined there in this view.

This translate to the LLIL fine:
```
tls_for_core:
   0 @ 0e035684  x1 = 0x40
   1 @ 0e035688  x0 = x0 * x1
   2 @ 0e03568c  x1 = tls_array
   3 @ 0e035690  x0 = x0 + x1
   4 @ 0e035694  <return> jump(x30)
```

However, going to MLIL is very wrong:
```
tls_for_core:
   0 @ 0e035694  return 
```

Part of this is due to the function signature being wrong. Originally its `void tls_for_core()`. If I update it to `void* tls_for_core(int core_num)` more weird things happen. The signature becomes `void*  tls_for_core(int32_t __saved_x0)`, and I am unable to rename the `__saved_x0` parameter. At this point the MLIL updates to:

```
tls_for_core:
   0 @ 0e035694  return __saved_x0
```

I'd expect to get something like:
```
tls_for_core:
  0 @ 0e035694  return &tls_array[num_core]
```

(or `tls_array + num_core * 0x40`, or similar)

**Comment by yrp604** [comment]
This is also the same bndb as https://github.com/Vector35/binaryninja-api/issues/2307#issue-825386754 if you need a test case.

**Comment by rssor** [comment]
Fixed in 2.4.2979-dev.
